src/utils/workspace.ts

```diff
@@ -144,7 +144,11 @@ export async function ensureFolderIsOpen(fsPath: string, actionContext: IActionC
 
         const uri: vscode.Uri = vscode.Uri.file(fsPath);
         if (openBehavior === OpenBehavior.AddToWorkspace) {
-            vscode.workspace.updateWorkspaceFolders(openFolders.length, 0, { uri: uri });
+            if (openFolders.length) {
+                vscode.workspace.updateWorkspaceFolders(openFolders.length, 0, { uri: uri });
+            } else {
+                await vscode.commands.executeCommand('vscode.openFolder', uri);
```

Comment:
nit: We try to avoid copied code as much as possible. I suggest you change the value of openBehavior before this if/else so that it will automatically call the existing code to 'OpenInCurrentWindow'